flow/raster_cache.cc

```diff
@@ -26,9 +26,8 @@ void RasterCacheResult::draw(SkCanvas& canvas, const SkPaint* paint) const {
   SkAutoCanvasRestore auto_restore(&canvas, true);
   SkIRect bounds =
       RasterCache::GetDeviceBounds(logical_rect_, canvas.getTotalMatrix());
-  FML_DCHECK(
-      std::abs(bounds.size().width() - image_->dimensions().width()) <= 1 &&
-      std::abs(bounds.size().height() - image_->dimensions().height()) <= 1);
+  FML_DCHECK(bounds.size().width() == image_->dimensions().width() &&
+             bounds.size().height() == image_->dimensions().height());
```

Comment:
This change is only valid if we have fractional metrics turned off. If they are turned on, then the <=1 threshold is needed.

Reply:
If the fractional offset is not turned off, then the fractional offsets must match exactly (see [RasterCacheKey][1]), so I think the rounded-out bounds should still exactly match the image size?

[1]: https://github.com/flutter/engine/blob/f9e53c72c656e240a1be969e38e989bb4667878a/flow/raster_cache_key.h#L18

Reply:
Ah, yes, I was thinking of the new code that you were working on and then had to revert. Under that case where we will cache all rasters with an offset of 0, then we can end up with a +1 on the image bounds. For the current code, then, this does make sense.